paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/pipeline/cdc/source/TableAwareFileStoreSourceSplit.java

```diff
@@ -0,0 +1,154 @@
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * A {@link FileStoreSourceSplit} that contains information about the table that corresponds to the
+ * {@link Split}.
+ */
+public class TableAwareFileStoreSourceSplit extends FileStoreSourceSplit {
+    private final Identifier identifier;
+    @Nullable private final TableSchema lastSchema;
+    private final TableSchema schema;
+
+    public TableAwareFileStoreSourceSplit(
+            String id,
+            Split split,
+            long recordsToSkip,
+            Identifier identifier,
+            @Nullable TableSchema lastSchema,
+            TableSchema schema) {
+        super(id, split, recordsToSkip);
+        this.identifier = identifier;
+        this.lastSchema = lastSchema;
+        this.schema = schema;
+    }
+
+    public Identifier getIdentifier() {
+        return identifier;
+    }
+
+    public @Nullable TableSchema getLastSchema() {
+        return lastSchema;
+    }
+
+    public TableSchema getSchema() {
+        return schema;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof TableAwareFileStoreSourceSplit)) {
+            return false;
+        }
+
+        TableAwareFileStoreSourceSplit other = (TableAwareFileStoreSourceSplit) o;
+        return Objects.equals(splitId(), other.splitId())
+                && Objects.equals(split(), other.split())
+                && recordsToSkip() == other.recordsToSkip()
+                && identifier.equals(other.identifier)
+                && Objects.equals(lastSchema, other.lastSchema)
+                && schema.equals(other.schema);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(splitId(), split(), recordsToSkip(), identifier, lastSchema, schema);
+    }
+
+    @Override
+    public String toString() {
+        return "TableAwareFileStoreSourceSplit{"
+                + "splitId='"
+                + splitId()
+                + "'"
+                + ", split="
+                + split()
+                + ", recordsToSkip="
+                + recordsToSkip()
+                + ", identifier="
+                + identifier
+                + ", lastSchema="
+                + lastSchema
+                + ", schema="
+                + schema
+                + '}';
+    }
+
+    /** The serializer for {@link TableAwareFileStoreSourceSplit}. */
+    public static class Serializer
+            implements SimpleVersionedSerializer<TableAwareFileStoreSourceSplit> {
+
+        @Override
+        public int getVersion() {
+            return 1;
+        }
+
+        @Override
+        public byte[] serialize(TableAwareFileStoreSourceSplit split) throws IOException {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            DataOutputViewStreamWrapper view = new DataOutputViewStreamWrapper(out);
+            view.writeUTF(split.splitId());
+            InstantiationUtil.serializeObject(view, split.split());
```

Comment:
I am concerned that this part may lead to state incompatibility due to modifications in Split. However, since `FileStoreSourceSplitSerializer` also implements this approach, I believe such incompatibility issues should be avoided through adjustments in Split.